docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+        console.log(`Searching for recipes with ${field} equal to ${value}`);
+        const body = {
+            'query': {
+                'term': {
+                    [field]: value
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js termSearch sodium 0
+
+
+Full-text queries
+^^^^^^^^^^^^^^^^^^
+
+Full-text queries returns results sorted by relevance. It allows higher flexibility to find most relevant results. For example, lets search for "Tomato garlic soup with dill" and look at the list of results.
+
+.. code-block:: javascript
+
+    /**
+     * Finding matches sorted by relevance.
+     */
+    module.exports.matchSearch = (field, query) => {
+        const body = {
+            'query': {
+                'match': {
+                    [field] : {
+                        query
+                    }
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js matchSearch title 'Tomato-garlic soup with dill'
+
+These are results which I got from the dataset:
+
+.. code-block:: text
+
+    [
+      'Mussel Soup with Avocado, Tomato, and Dill ',
+      'Garlic Soup ',
+      'Cucumber-Dill Soup with Scallions ',
+      'Cold Tomato-Thyme Soup with Grilled Garlic Croutons ',
+      'Chilled Tomato, Roasted Garlic, and Basil Soup ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Asparagus and Dill Avgolemono Soup ',
+      'Garlic Tomato Sauce ',
+      'Creamy Tomato Soup '
+    ]
+
+OpenSearch engine returned 10 most relevant results. Why 10? Because it is a default value. It can be increased by setting size property to a higher number.
+
+Next, lets use some special operators in a search query with *a query string*. Let's also increase the number of returned results to 100 to demonstrate how we can get more than 10 values.
+
+.. code-block:: javascript
+
+    /**
+     * Using special operators within a query string and a size parameter.
+     */
+    module.exports.querySearch = (query, size) => {
+        console.log(`Searching for ${query} and returning maximum ${size} results`);
+        const body = {
+            'query': {
+                'query_string': {
+                    "fields" : ["ingredients"],
+                    query
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body,
+            size
+        }, logTitles)
+    };
+
+The full list of operators and their shortcuts can be found `in the documentation <https://opensearch.org/docs/opensearch/query-dsl/full-text/#simple-query-string>`_.
+
+To find recipes with tomato, salmon or tuna and no onion run the next query:
+
+::
+
+    run-func index.js querySearch "(salmon|tuna) +tomato -onion" 100
+
+
```

Comment:
Invite people to try the query with their own likes/dislikes? More words in between the code samples helps to make it digestible (pun entirely intended)

Do we need a title here too, before the next example?